draft-ietf-mls-protocol.md

```diff
@@ -1003,69 +1012,119 @@ enum {
 } ContentType;
 
 struct {
-    opaque content<0..2^32-1>;
-    opaque signature<0..2^16-1>;
+    opaque content[MLSPlaintext.length];
+    uint8 signature[MLSInnerPlaintext.sig_len];
+    uint16 sig_len;
+    uint8  marker;
     uint8  zero\_padding[length\_of\_padding];
-} Message;
+} MLSInnerPlaintext;
 
 struct {
-    ContentType type;
-    select (Plaintext.type) {
+    select (MLSPlaintext.type) {
         case handshake:   Handshake;
         case application: Application;
     }
-} Plaintext;
+} MLSPlaintext;
```

Comment:
I would prefer if this struct had everything that was not necessary for encryption.  That way if you're not encrypting Handshake messages, you just send this struct instead.  So I would add back:

* Group ID
* Content type
* Signature